scripts/update_institution_sso_email_domain.py

```diff
@@ -0,0 +1,117 @@
+    errored_users = []
+    for user in affiliated_users:
+        result = update_email_domain(user, src_domain, dst_domain)
+        if result == UpdateResult.SUCCEEDED:
+            succeeded_users.append(user._id)
+        if result == UpdateResult.SKIPPED:
+            skipped_users.append(user._id)
+        if result == UpdateResult.FAILED:
+            failed_users.append(user._id)
+        if result == UpdateResult.ERRORED:
+            errored_users.append(user._id)
+
+    # Output update results to console
+    logger.info(f'{UpdateResult.SUCCEEDED.name} ({len(succeeded_users)}/{len(affiliated_users)}) = {succeeded_users}')
+    logger.warning(f'{UpdateResult.SKIPPED.name} ({len(skipped_users)}/{len(affiliated_users)}) = {skipped_users}')
+    logger.error(f'{UpdateResult.FAILED.name} ({len(failed_users)}/{len(affiliated_users)}) = {failed_users}')
+    logger.error(f'{UpdateResult.ERRORED.name} ({len(errored_users)}/{len(affiliated_users)}) = {errored_users}')
+
+
+def update_email_domain(user, src_domain, dst_domain):
+    """For a given user, if it has an email `example@<src_domain>`, attempt to add `example@<dst_domain>` to this
+    account. The action will be skipped if `example@<dst_domain>` already exists on this user or no email is found
+    under `@<src_domain>`. The action will fail if `example@<dst_domain>` already exists on another account. Other
+    unexpected exceptions will be considered as error.
+    """
+
+    # Find the eligible email to add
+    email_to_add = None
+    for email in user.emails.all():
+        email_parts = email.address.split('@')
+        name_part = email_parts[0].lower()
+        domain_part = email_parts[1].lower()
+        if domain_part == src_domain:
+            email_to_add = f'{name_part}@{dst_domain}'
+            break
+    if not email_to_add:
+        logger.warning(f'Action skipped due to no eligible email found for user [{user._id}]!')
+        return UpdateResult.SKIPPED
+    # Verify and attempt to add email; keep track of successes, failures and errors
+    duplicate_user = get_user(email=email_to_add)
+    if not duplicate_user:
+        try:
+            user.emails.create(address=email_to_add)
+            user.save()
+        except Exception:
+            logger.error(f'An error occurred when adding email [{email_to_add}] to user [{user._id}]!')
+            sentry.log_exception()
+            return UpdateResult.ERRORED
+        logger.info(f'Successfully added email [{email_to_add}] to user [{user._id}]!')
+        return UpdateResult.SUCCEEDED
+    elif duplicate_user == user:
+        logger.info(f'Action skipped since email [{email_to_add}] exists for the same user [{user._id}]!')
+        return UpdateResult.SKIPPED
+    else:
+        logger.info(f'Action aborted since email [{email_to_add}] exists '
+                    f'for a different user user [{duplicate_user._id}]!')
+        return UpdateResult.FAILED
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
Please make this a management command instead; `/scripts/` is generally deprecated and not preferred for new work that may need to be ran multiple times.

Reply:
Good catch, will move this to the management command.

Reply:
Done